Add html validation to radio buttons on category tree

NeOMakinG opened the pull request:
<!-----------------------------------------------------------------------------
Thank you for contributing to the PrestaShop project! 

Please take the time to edit the "Answers" rows below with the necessary information.

Check out our contribution guidelines to find out how to complete it:
https://devdocs.prestashop.com/1.7/contribute/contribution-guidelines/#pull-requests
------------------------------------------------------------------------------>

| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Category trees are not blocking the submit form if required
| Type?             | bug fix
| Category?         | BO
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | Fixes #19612.
| How to test?      | See issue, btw, it doesn't fix the material checkbox one, that should be done using JavaScript inside the UIKit, with an helper class in order to create the control : 1. Install PS and [sample module (https://github.com/202ecommerce/checkboform) 2. See error by validating the form without checking checkboxes (won't work on material checkboxes)
| Possible impacts? | Every radio trees


<!-- Click the form's "Preview" button to make sure the table is functional in GitHub. Thank you! -->

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/prestashop/prestashop/22942)
<!-- Reviewable:end -->

⚠️ Mergeable when https://github.com/PrestaShop/prestashop-ui-kit/pull/137 is merged and UIKit 1.2.2 is released ⚠️

Commit:
Add html validation to radio buttons on category tree

Commit:
Remove useless space

Commit:
Remove required on checkbox which cant be validated by the browser

Commit:
Remove useless space

Commit:
Add required class to material choice tree for JS

matks reviewed (dismissed):
Code seems good but could you please make it a little more clear what is fixed, what is not, and how QA can validate this PR 😉

khouloudbelguith commented:
Hi @NeOMakinG,

I have only one issue in the category page when multistore enabled.
Steps to reproduce the issue:
1. Enable Multistore Context
2. Create a new shop
3. Go to BO > Catalog > Categories page
4. Add a new root category > Save 
5. At this step, you are in the Root category list 
![image](https://user-images.githubusercontent.com/16067358/105733633-0dd50580-5f32-11eb-961e-f2b79ab88481.png)
6. Add a new category > the Parent category is not selected
7. Set only the Name & save
8. See the error => nothing happens

I attached a screen record
https://drive.google.com/file/d/11PXqvCzvJq0lrAcG4DW0eRn97kjfoUFE/view?usp=sharing

**Without PR**
When I click save => I have an alert displayed
![image](https://user-images.githubusercontent.com/16067358/105734051-7fad4f00-5f32-11eb-86fa-e879ae4b2eb1.png)

Thank you!

NeOMakinG commented:
@khouloudbelguith 

Thanks for testing, this error isn't related at all, it's just because the form doesn't submit as you didn't fill the category tree, btw I shouldn't block these fields as they can't be fixed only by doing this, I need to go further on this PR.

NeOMakinG commented:
@khouloudbelguith could you test again ? I checkout the PR and it looks like I can't reproduce your behavior at all !

Commit:
Add text danger on category parent of create category page

NeOMakinG commented:
I added the "required" little asterisk and fixed this problem on UIKit side : https://github.com/PrestaShop/prestashop-ui-kit/pull/137

It's testable with this PR as well

PierreRambaud commented on src/PrestaShopBundle/Resources/views/Admin/TwigTemplateForm/bootstrap_4_layout.html.twig:
```diff
@@ -200,7 +201,7 @@
             <div class="radio">
                 <label>
-                    <input type="radio" name="form[{{ form.vars.id }}][tree]" value="{{ child.id_category }}" {{ checked }} class="category">
+                    <input type="radio" name="form[{{ form.vars.id }}][tree]" value="{{ child.id_category }}" {{ checked }} class="category" {% if required %}required{% endif %}>
```
```suggestion
                    <input type="radio" name="form[{{ form.vars.id }}][tree]" value="{{ child.id_category }}" {{ checked }} class="category"{% if required %} required{% endif %}>
```

PierreRambaud commented on src/PrestaShopBundle/Resources/views/Admin/TwigTemplateForm/material.html.twig:
```diff
@@ -24,7 +24,7 @@
 
 {% block material_choice_tree_widget %}
-<div class="material-choice-tree-container js-choice-tree-container" id="{{ form.vars.id }}">
+  <div class="material-choice-tree-container js-choice-tree-container {% if required %}required{% endif %}" id="{{ form.vars.id }}">
```
```suggestion
  <div class="material-choice-tree-container js-choice-tree-container{% if required %} required{% endif %}" id="{{ form.vars.id }}">
```

PierreRambaud reviewed (changes requested):
Just two comments :)

PierreRambaud commented on src/PrestaShopBundle/Resources/views/Admin/TwigTemplateForm/bootstrap_4_layout.html.twig:
```diff
@@ -186,7 +186,8 @@
             <div class="checkbox">
                 <label>
-                    <input type="checkbox" name="{{ form.vars.full_name }}[tree][]" value="{{ child.id_category }}" class="category" {{ checked }}>
+                  <input type="checkbox" name="{{ form.vars.full_name }}[tree][]" value="{{ child.id_category }}" class="category" {{ checked }}>
```
```suggestion
                    <input type="checkbox" name="{{ form.vars.full_name }}[tree][]" value="{{ child.id_category }}" class="category" {{ checked }}>
```

khouloudbelguith commented:
Hi @NeOMakinG,

Using this PR: https://github.com/PrestaShop/prestashop-ui-kit/pull/137, I found only an error in the console
Steps to reproduce the issue:

1. Enable Multistore Context
2. Create a new shop
3. Go to BO > Catalog > Categories page
4. Add a new root category > Save
5. At this step, you are in the Root category list
6. Select all only the `first shop`/`second shop`
7. I have an error but it is only on the console
8. Add a new category > the Parent category is not selected
9. Set only the Name & save
10. See the error on the console

![image](https://user-images.githubusercontent.com/16067358/108342563-3a033f80-71db-11eb-923a-eb8242135d37.png)

What do you think?

Thank you!

NeOMakinG commented:
@khouloudbelguith 

Thanks a lot for testing.

I understand this error, it's due to radio elements which are inside a `display: none;` property. so the browser is unable to show the error message on an hidden element.

I think it's fine enough for this work, overwise it would need a lot of javascript in order to handle every forms, which could lead to side effects. Or a rework of radio buttons which is a BC...

Commit:
Fix condition in form

Commit:
Fix condition in form

Commit:
Fix defaultHidden condition

Progi1984 reviewed (commented):
Please fix review from @PierreRambaud

Commit:
Fix review erased

NeOMakinG commented:
@Progi1984 idk why it has been removed, but I fixed it again :)

PierreRambaud committed:
Fix indentation in bootstrap template

PierreRambaud committed:
Fix unneeded space

khouloudbelguith commented:
Hi @NeOMakinG,

An issue is created here: https://github.com/PrestaShop/PrestaShop/issues/23365

It is ok :heavy_check_mark: 

**With PR**
![image](https://user-images.githubusercontent.com/16067358/108876584-ab3d5b00-75fe-11eb-8884-cb2cc5c0f11a.png)
**Without PR**
![image](https://user-images.githubusercontent.com/16067358/108876735-d32cbe80-75fe-11eb-88bc-64ce7c7ed2ec.png)

Thank you!

Progi1984 reviewed (changes requested):
## WAITING FOR RELEASE OF UIKit 1.2.2

Progi1984 commented:
Thanks @NeOMakinG & @khouloudbelguith
